### PR TITLE
Documentation update after removal of `kyma CLI` invocations

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

Changes proposed in this pull request:

- Release workflow changed
- Testing flow changed for all E2E tests

**Related issue(s)**

### CODE REVIEW

File: docs/assets/release.svg
Comment (IwonaLanger) [inline docs/assets/release.svg:1]: Please change the background for "Create a release" back to blue or keep it white. If you're unsure which one is better, see our [guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/09-diagrams.md#background).

File: docs/contributor/03-10-release.md [hunk 1]
```diff
@@ -22,18 +21,17 @@ To create a release, follow these steps:
    iv. provide a version, for example, 1.2.0  
    v. choose real or dummy credentials for Service Manager  
    vi. choose whether to bump or not to bump the security scanner config
```
Comment (IwonaLanger) [inline docs/contributor/03-10-release.md:23]:
```suggestion
   vi. choose whether to bump or not to bump the security scanner config  
```

File: docs/contributor/03-10-release.md [hunk 2]
```diff
@@ -22,18 +21,17 @@ To create a release, follow these steps:
...
 10. The GitHub action asynchronously initiates stress tests jobs and E2E tests jobs upon Prow job success status. E2E upgrade tests run only with real credentials for Service Manager.
-11. The GitHub action waits for the `template.yaml` asset in the GitHub release and for images in the Docker registry.
-12. The GitHub action fetches the module image and runs E2E tests in parallel on the k3s clusters for the most recent k3s versions and with the specified credentials. The number of the most recent k3s versions to be used is defined in the **vars.LAST_K3S_VERSIONS** GitHub variable. 
-13. If the unit tests, stress tests, and E2E tests are completed successfully, the GitHub action publishes the release.
+12. The GitHub action runs E2E tests in parallel on the k3s clusters for the most recent k3s versions and with the specified credentials. The number of the most recent k3s versions to be used is defined in the **vars.LAST_K3S_VERSIONS** GitHub variable. 
+13. If the unit tests, stress tests, and E2E tests are completed successfully (and you did not choose to not publish in 1.vii), the GitHub action publishes the release.
```
Comment (IwonaLanger) [inline docs/contributor/03-10-release.md:34]:
```suggestion
13. If the unit tests, stress tests, and E2E tests are completed successfully and you have chosen to publish in 1.vii, the GitHub action publishes the release.
```

File: docs/assets/release.svg
Comment (jaroslaw-pieszka) [reply docs/assets/release.svg:1]: Should be fixed in the last commit

File: docs/contributor/03-10-release.md [hunk 3]
```diff
@@ -21,19 +20,18 @@ To create a release, follow these steps:
...
    iv. provide a version, for example, 1.2.0  
    v. choose real or dummy credentials for Service Manager  
-   vi. choose whether to bump or not to bump the security scanner config
+   vi. choose whether to bump or not to bump the security scanner config  
+   vii. choose whether you want to publish the release
 2. The GitHub action, defined in the [`create-release.yaml`](/.github/workflows/create-release.yaml) file, validates the release by checking if the GitHub tag already exists, if there are any old Docker images for that GitHub tag, and if merged PRs that will be part of this release are labeled correctly.
 3. If you chose in step 1.vi to bump the security scanner config, the GitHub action creates a PR with a new security scanner config that includes the new GitHub tag version.
 4. A code owner approves the PR. 
-5. The GitHub action creates a GitHub tag and draft release with the provided name.
+5. The GitHub action creates a GitHub tag and draft release with the provided name, creates release assets and uploads these.
```
Comment (IwonaLanger) [inline docs/contributor/03-10-release.md:28]:
```suggestion
5. The GitHub action creates a GitHub tag and draft release with the provided name; then, it creates release assets and uploads them.
```

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**